umbral-pre-wasm/patches/api-compatibility.js.txt

```diff
@@ -0,0 +1,40 @@
+
+/*
+    contents added by umbral-pre-wasm makefile/build script
+*/
+
+import { KeyFrag } from  "./umbral_pre_wasm_bg.js";
+
+export const decrypt_reencrypted = (
+    receiving_sk,//: SecretKey,
+    delegating_pk,//: PublicKey,
+    capsule,//: Capsule,
+    verified_cfrags,//: Sequence[VerifiedCapsuleFrag],
+    ciphertext//: bytes,) => (
+) => {
+    const capsuleWithCFrags = verified_cfrags.reduce((value, cfrag) => {
+        return value.withCFrag(cfrag)
+    }, capsule)
+    return capsuleWithCFrags.decryptReencrypted(receiving_sk, delegating_pk, ciphertext);
+}
+
+KeyFrag.prototype.verify =  (verifying_pk, optional = {delegating_pk, receiving_pk} = {}) => {
```

Comment:
Interesting construction here - `optional = {delegating_pk, receiving_pk} = {}`. Does it means that it checks that the given dictionary can only have keys with these names?

Reply:
Yeah so you'd call it like
`kfrag.verify(pk)` in which case the default value of {} would be passed in.
but you could call it like `kfrag.verify(pk, {delegating_pk, receiving_pk})`

the nice thing is that inside the method,  testing `optional.delegating_pk` will work.

I think @piotr-roslaniec [came up with this](https://github.com/nucypher/rust-umbral/issues/25#issuecomment-883579430) if I remember correctly.